maps/reddit.py

```diff
@@ -0,0 +1,111 @@
+import time
+
+def get_reddit_instance(client_id, client_secret, user_agent):
+    reddit = praw.Reddit(
+        client_id=client_id,
+        client_secret=client_secret,
+        user_agent=user_agent
+    )
+    return reddit
+
+@click.command()
+@click.option('--client-id', prompt='Enter your Reddit client ID', help='Reddit client ID')
+@click.option('--client-secret', prompt='Enter your Reddit client secret', hide_input=True, help='Reddit client secret')
+@click.option('--user-agent', prompt='Enter your Reddit user agent', help='Reddit user agent')
+@click.option('--reddit-url', prompt='Enter Reddit post URL', help='URL of the Reddit post')
+@click.option('--nomic-api-key', prompt='Enter your Nomic API', help='Nomic API key')
+def main(client_id, client_secret, user_agent, reddit_url, nomic_api_key):
+    reddit = get_reddit_instance(client_id, client_secret, user_agent)
+
+    submission = reddit.submission(url=reddit_url)  # Use submission URL directly
+
+    comments = scrape_reddit_comments(reddit, submission)
+
+    # Convert comments to Apache Arrow table
+    arrow_table = comments_to_arrow_table(comments)
+
+    try:
+        # Generate map name based on Reddit post title
+        map_name = f"Reddit_Post_{submission.title}"
+
+        dataset = atlas.map_data(data=arrow_table,
+                                 indexed_field='text',
+                                 description='Reddit comments mapped via automation.',
+                                 topic_model=True,
+                                 map_name=map_name  # Use generated map name
+                                )
+        if dataset and 'id' in dataset:
+            print("Map created on Atlas with ID:", dataset['id'])
+            print("All done! Visit the map link to see the status of your map build.")
+        else:
+            print("Map creation failed. Dataset ID not found in response.")
+    except Exception as e:
+        print(f"An error occurred while building map on Atlas: {e}")
+
+def scrape_reddit_comments(reddit, submission):
+    max_attempts = 5  
+    current_attempt = 0
+    while current_attempt < max_attempts:
+        try:
+            submission.comments.replace_more(limit=None)
+            comments = fetch_all_comments(submission)
+            print(f"Number of comments fetched: {len(comments)}")
+            return comments
+        except PrawcoreException as e:
+            if e.response and e.response.status_code == 429:
+                delay = 2 ** current_attempt  
+                print(f"Rate limit exceeded. Retrying in {delay} seconds...")
+                time.sleep(delay)
+                current_attempt += 1
+            else:
+                print(f"Error scraping comments: {e}")
+                return []
+
+    print("Max retry attempts reached. Could not fetch comments.")
+    return []
+
+def fetch_all_comments(submission):
+    comments = []
+    submission.comments.replace_more(limit=None)
```

Comment:
Is this going to get _all_ top level comments? I think I see how the replies will be recursively plumbed but I'm not sure if we'll have all the top comments to start.

Reply:
from what ive tested so far it does, but maybe larger datasets will cause issues

Reply:
from what ive tested so far it does, but maybe larger datasets will cause issues